tests/utils/test_embedding_resize.py

```diff
@@ -197,3 +206,41 @@ def test_resize_with_multiple_of():
     assert model.get_input_embeddings().embedding_dim % 8 == 0
     assert resize_result["new_embedding_size"] % 8 == 0
     assert model.get_output_embeddings().out_features % 8 == 0
+
+
+def test_resize_llama_vision_model():
+    model = AutoModelForVision2Seq.from_pretrained(LLAMA_VISION_MODEL_NAME)
+    processor = AutoProcessor.from_pretrained(LLAMA_VISION_MODEL_NAME)
+    tokenizer = processor.tokenizer
+
+    current_input_embeddings = model.get_input_embeddings()
+    current_input_embeddings = copy.deepcopy(current_input_embeddings)
+    current_output_embeddings = model.get_output_embeddings()
+    current_output_embeddings = copy.deepcopy(current_output_embeddings)
+
+    current_tokenizer_len = len(tokenizer.get_vocab())
+
+    resize_result = tokenizer_and_embedding_resize(
+        special_tokens_dict={"unk_token": "<unk>"},
+        tokenizer=tokenizer,
+        model=model,
+        multiple_of=1,
+    )
+
+    resized_input_embeddings = model.get_input_embeddings()
+    resized_output_embeddings = model.get_output_embeddings()
+    resized_tokenizer_len = len(tokenizer.get_vocab())
+
+    assert resized_tokenizer_len == current_tokenizer_len + 1
+    assert "<unk>" in tokenizer.get_vocab()
+    assert resize_result["num_new_tokens"] == 1
+
+    # 2 new tokens were added: <unk> and <image>
+    assert (
```

Comment:
@Abhishek-TAMU can you please add 2 lines from your comments here [1](https://github.com/foundation-model-stack/fms-hf-tuning/pull/451#discussion_r2025456404) and [2](https://github.com/foundation-model-stack/fms-hf-tuning/pull/451#discussion_r2025475597) which mention why have we added `2` new tokens.